apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -65,6 +70,43 @@ defmodule GameBackend.CurseOfMirra.Quests do
     Quest.changeset(quest, attrs)
   end
 
+  def get_daily_quest(daily_quest_id) do
+    q = from(dq in DailyQuest, preload: [:quest], where: dq.id == ^daily_quest_id)
+
+    Repo.one(q)
+  end
+
+  def get_users_daily_quests(user_id) do
+    q = from(dq in DailyQuest, preload: [:quest], where: dq.user_id == ^user_id)
+
+    Repo.all(q)
+  end
+
+  @doc """
+  Get all %DailyQuest{} for the user_id that were inserted today and were rerolled
+
+  ## Examples
+
+      iex>get_user_today_rerolled_daily_quests_by_type(user_id, "daily")
+      [%Quest{type: "daily"}]
+  """
+  def get_user_today_rerolled_daily_quests_by_type(user_id, type) do
+    naive_today = NaiveDateTime.utc_now()
+    start_of_date = NaiveDateTime.beginning_of_day(naive_today)
+    end_of_date = NaiveDateTime.end_of_day(naive_today)
+
+    q =
+      from(dq in DailyQuest,
```

Comment:
If you're taking only DailyQuest into account, why do you have to explicit the type in the function? I think we should either remove the type argument or make the pattern-matching functions for each type.
Also remember to fix the @doc and function name if you change this.

Reply:
yeah I think the best approach is to pattern match the quest  type and when we have more quest types like "weekly" or "match" create a proper match to retrieve them